weaver/wps_restapi/jobs/jobs.py

```diff
@@ -582,11 +586,97 @@ def get_job_outputs(request):
+    response_schemas=sd.get_prov_result_responses
+)
+@sd.process_output_service.get(
+    tags=[sd.TAG_JOBS, sd.TAG_RESULTS, sd.TAG_PROCESSES],
+    schema=sd.ProcessAnyOutputEndpoint(),
+    response_schemas=sd.get_proc_result_responses
+)
+@sd.job_output_service.get(
+    tags=[sd.TAG_JOBS, sd.TAG_RESULTS, sd.TAG_PROCESSES],
+    schema=sd.JobAnyOutputEndpoint(),
+    response_schemas=sd.get_job_result_responses
+)
+@log_unhandled_exceptions(logger=LOGGER, message=sd.InternalServerErrorResponseSchema.description)
+def get_job_output(request):
+    # type: (PyramidRequest) -> AnyResponseType
+    """
+    Retrieve a specific output from a job execution.
+    """
+    job = get_job(request)
+    raise_job_dismissed(job, request)
+    raise_job_bad_status_success(job, request)
+    settings = get_settings(request)
+    output_id = request.matchdict.get("output_id")
+    # Get requested media-type. "*/*" if omit
+    accept = str(request.accept) if request.accept else "*/*"
+    headers = request.headers
+    results = [o for o in job.results if str(o["identifier"]) == output_id]
+    if results:
+        result = results[0]
+        mime_type = get_field(result, "mime_type", search_variations=True, default="")
+        possible_media_types = transform.CONVERSION_DICT.get(mime_type, [])
+        possible_media_types.append(mime_type)
+    else:
+        raise HTTPNotFound(
+            json={
+                "code": "",
+                "description": "The requested output Id is not available in the job results.",
+                "cause": "The output ID is not available",
+                "error": "",
+                "value": ""
+            }
+        )
+    result_media_type = get_field(result, "mimeType", search_variations=True)
+    result_media_type = guess_target_format(request, default=result_media_type)
+
+    if result_media_type not in possible_media_types:
+        raise HTTPUnprocessableEntity(json={
+            "code": "InvalidMimeTypeRequested",
+            "description": "The requested output format is not in the possible output formats.",
+            "cause": "Incompatible mime Types",
+            "error": "InvalidMimeTypeRequested",
+            "value": ""
+        })
+
+    is_reference = bool(get_any_value(result, key=True, file=True))
+    _, output_format = get_job_output_transmission(job, output_id, is_reference)
+    output_format = accept or output_format or result_media_type
+
+    # To ensure consistency and avoid type mismatches, all output formats are converted to a dictionary.
+    # This standardization prevents errors when handling cases in `get_job_results_single`
+    # where some formats are dictionaries and others are different types (e.g., strings or ContentType).
+    if not isinstance(output_format, dict):
+        output_format = {"mime_type": output_format}
+
+    return get_job_results_single(job, result, output_id, output_format, headers=headers, settings=settings)
```

Comment:
This is essentially the same as the end of `get_job_results_response`: 

https://github.com/crim-ca/weaver/blob/64db3485b2dbb808a5d75e1b63a586cfa1e21147/weaver/wps_restapi/jobs/utils.py#L758-L765

Please define a function like `resolve_result_single` or similar, and invoke it in both places. 

The output transmission check should already be done by the `get_job_results_single` call, with all corresponding logic to convert data/link and transforms.